primers/pcr/pcr.go

```diff
@@ -0,0 +1,138 @@
+package pcr
+
+import (
+	"github.com/TimothyStiles/poly/primers"
+	"github.com/TimothyStiles/poly/transform"
+	"index/suffixarray"
+	"sort"
+	"strings"
+)
+
+// https://doi.org/10.1089/dna.1994.13.75
+var minimalPrimerLength int = 15
+
+func DesignPrimersWithOverhangs(sequence, forwardOverhang, reverseOverhang string, targetTm float64) (string, string) {
```

Comment:
Needs comment for docs.

Reply:
Can you also give some overview on how primer design works? What are overhangs and why are they needed?, etc.